docs/guides/go.md

```diff
@@ -311,112 +310,89 @@ information on why the transaction was rejected.
 
 Note that `CheckTx` does not execute the transaction, it only verifies that that the transaction could be executed. We do not know yet if the rest of the network has agreed to accept this transaction into a block.
 
-
 Finally, make sure to add the bytes package to the `import` stanza at the top of `app.go`:
 
 ```go
 import(
-	"bytes"
+    "bytes"
 
-	"github.com/dgraph-io/badger/v3"
-	abcitypes "github.com/cometbft/cometbft/abci/types"
+    "github.com/dgraph-io/badger/v3"
+    abcitypes "github.com/cometbft/cometbft/abci/types"
 )
 ```
 
+### 1.3.3 FinalizeBlock
 
-### 1.3.3 BeginBlock -> DeliverTx -> EndBlock -> Commit
-
-When the CometBFT consensus engine has decided on the block, the block is transferred to the
-application over three ABCI method calls: `BeginBlock`, `DeliverTx`, and `EndBlock`.
-
-- `BeginBlock` is called once to indicate to the application that it is about to
-receive a block.
-- `DeliverTx` is called repeatedly, once for each application transaction that was included in the block.
-- `EndBlock` is called once to indicate to the application that no more transactions
-will be delivered to the application in within this block.
+`FinalizeBlock` is a method introduced in CometBFT v0.38.0 which replaces the traditional ABCI methods of `BeginBlock`, `DeliverTx`, and `EndBlock`.
 
-Note that, to implement these calls in our application we're going to make use of Badger's
-transaction mechanism. We will always refer to these as Badger transactions, not to
-confuse them with the transactions included in the blocks delivered by CometBFT,
-the _application transactions_.
```

Comment:
This paragraph is important not to confuse the reader with the two different meanings of the term "transaction" (database transactions vs. blockchain transactions). Is it kept somewhere?